frontend/src/components/MeetingRoom/DeviceControlButton/DeviceControlButton.tsx

```diff
@@ -0,0 +1,109 @@
+import Mic from '@mui/icons-material/MicNone';
+import { IconButton } from '@mui/material';
+import VideocamIcon from '@mui/icons-material/Videocam';
+import VideocamOffIcon from '@mui/icons-material/VideocamOff';
+import Tooltip from '@mui/material/Tooltip';
+import ButtonGroup from '@mui/material/ButtonGroup';
+import { MicOff, ArrowDropUp, ArrowDropDown } from '@mui/icons-material';
+import { useState, useRef, useCallback, ReactElement } from 'react';
+import MutedAlert from '../../MutedAlert';
+import usePublisherContext from '../../../hooks/usePublisherContext';
+import DeviceSettingsMenu from '../DeviceSettingsMenu';
+
+export type DeviceControlButtonProps = {
+  isAudioControl?: boolean;
+};
+
+/**
+ * DeviceControlButton Component
+ *
+ * This component displays a current status of audio/video device (camera/microphone enabled/disabled)
+ * and shows a dropdown that displays available audio/video devices.
+ * @param {DeviceControlButtonProps} props - the props for the component.
+ *  @property {boolean} isAudioControl - (optional) indicates whether a device is an audio control, if set to false it is a video control.
```

Comment:
In general I'd say I'm against boolean props telling a component what it is, I'd rather just render shared components or have shared hooks.
Also if we need to have a prop it can be something like type="video", rather than `isAudioControl` false meaning video.

I'll leave this for your judgement though

Reply:
I think I'll go with the second option here of using `type` set to either `audio` or `video` here. hope that's okay 🙏